File: ClavaWeaver/resources/clava/test/weaver/cpp/src/function.cpp
#include "function.h"

inline int Max(int x, int y) {
   return (x > y)? x : y;
}

int MaxNoInline(int x, int y) {
   return (x > y)? x : y;
}

int declOnly();

int defOnly() {
	return 1;
}

int declAndDef();

int declAndDef() {
	return 2;
}



void caller() {
	declOnly();
	defOnly();
	declAndDef();
	
	A a;
	a.declOnly();
}

File: ClavaWeaver/resources/clava/test/weaver/Function.js
laraImport("weaver.Query");

for (const $function of Query.search("function")) {
    const fName = $function.name;
    if (fName != "Max" && fName != "MaxNoInline") {
        continue;
    }

    console.log($function.name + " is inline: " + $function.isInline);
}

// Set name
for (const $function of Query.search("file", "function.cpp").search(
    "function"
)) {
    const fName = $function.name;
    if (
        fName != "defOnly" &&
        fName != "declOnly" &&
        fName != "declAndDef" &&
        fName != "notCalled"
    ) {
        continue;
    }

    //console.log("FILE BEFORE SET:\n" + $file.code);
    const newName = fName + "New";
    $function.name = newName;
    //console.log("FILE AFTER SET:\n" + $file.code);

    console.log("Function new name: " + $function.name);
}

for (const $function of Query.search("function", "caller")) {
    console.log("Caller body: " + $function.body.code);
}
